soda_machine.rb

```diff
@@ -1,18 +1,32 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_accessor :sodas, :cash, :sold_soda
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      end
+    end
+    nil
   end
 
-  def sell(soda_brand)
```

Comment:
This is super complicated. Please refactor this method and try to leverage the `find_soda` method already written to simplify the logic.